cluster-autoscaler/utils/drain/drain_test.go

```diff
@@ -588,5 +611,12 @@ func TestDrain(t *testing.T) {
 		if len(pods) != len(test.expectPods) {
 			t.Fatalf("Wrong pod list content: %v", test.description)
 		}
+
+		if len(daemonSetPods) != len(test.expectDaemonSetPods) {
```

Comment:
I think you can replace both these ifs with assert.ElementsMatch(t, test.expectDaemonSetPods, daemonSetPods).